### ISSUE TITLE
[Bug]: Compression job not restarting when paused or stopped forcefully

### ISSUE THREAD

**Issue by RobAtticus** [timescale/timescaledb#5978]
### What type of bug is this?

Other

### What subsystems and features are affected?

Background worker, Compression

### What happened?

Several times now I've had to stop a compression job using `SELECT alter_job(<id>, scheduled=>FALSE);` or have had the job stopped forcefully for other reasons, and afterwards the job will not restart.

I've tried using `SELECT alter_job(<id>, scheduled=>TRUE)` or sometimes include a `next_start` in the near future (seconds/minutes) but never seem to get the job restarted. Other jobs continue to run and schedule as normal, even if they too are forcefully stopped (e.g. a server shutdown). It only seems to be compression. Eventually I just remove and re-add the job.

### TimescaleDB version affected

2.11.1

### PostgreSQL version used

15

### What operating system did you use?

Cloud

### What installation method did you use?

Not applicable

### What platform did you run on?

Timescale Cloud

### How can we reproduce the bug?

```bash
Unsure if it's just me, but basically:

1. Add a compression job
2. Wait for it to be running
3. SELECT alter_job(<id>, scheduled=>FALSE); or otherwise stop it
4. SELECT alter_job(<id>, scheduled=>TRUE, next_start=>now() + '10s'::INTERVAL); and see if it restarts
```

**Comment by mkindahl** [comment]
@RobAtticus Thanks for the bug report. Unfortunately I cannot reproduce it when following the instructions above (see transcript below). 

It would help if we could see the log for the scheduler process and the policy job around the time when the compression policy stopped the first time, and from the scheduler around when it is supposed to start.

If you can also run the compression manually once and make sure that there is no other problem making the compression not work.

I assume that you're checking the `next_start` for the job to decide if the job has executed, like this:

```sql
SELECT job_id, application_name, next_start FROM timescaledb_information.jobs where job_id = :job_id;
```

Transcript of my run:
```
mats@fury:~/issues$ psql -f issue-5978.sql
Expanded display is used automatically.
Null display is "[NULL]".
CREATE EXTENSION IF NOT EXISTS timescaledb;
psql:issue-5978.sql:3: WARNING:  
WELCOME TO
 _____ _                               _     ____________  
|_   _(_)                             | |    |  _  \ ___ \ 
  | |  _ _ __ ___   ___  ___  ___ __ _| | ___| | | | |_/ / 
  | | | |  _ ` _ \ / _ \/ __|/ __/ _` | |/ _ \ | | | ___ \ 
  | | | | | | | | |  __/\__ \ (_| (_| | |  __/ |/ /| |_/ /
  |_| |_|_| |_| |_|\___||___/\___\__,_|_|\___|___/ \____/
               Running version 2.12.0-dev
For more information on TimescaleDB, please visit the following links:

 1. Getting started: https://docs.timescale.com/timescaledb/latest/getting-started
 2. API reference documentation: https://docs.timescale.com/api/latest
 3. How TimescaleDB is designed: https://docs.timescale.com/timescaledb/latest/overview/core-concepts

Note: TimescaleDB collects anonymous reports to better understand and assist our users.
For more information and how to disable, please see our docs https://docs.timescale.com/timescaledb/latest/how-to-guides/configuration/telemetry.

CREATE EXTENSION
SELECT remove_compression_policy('conditions', if_exists => true);
psql:issue-5978.sql:5: ERROR:  relation "conditions" is not a hypertable or continuous aggregate
DROP TABLE IF EXISTS conditions;
DROP TABLE
CREATE TABLE conditions(
    time TIMESTAMPTZ NOT NULL,
    device INTEGER,
    temperature FLOAT
);
CREATE TABLE
SELECT * FROM create_hypertable('conditions', 'time');
 hypertable_id | schema_name | table_name | created 
---------------+-------------+------------+---------
             1 | public      | conditions | t
(1 row)

INSERT INTO conditions
SELECT time, (random()*30)::int, random()*80 - 40
FROM generate_series(NOW() - INTERVAL '6 days', NOW(), '1 minute') AS time;
INSERT 0 8641
ALTER TABLE conditions SET (
    timescaledb.compress,
    timescaledb.compress_segmentby = 'device'
);
ALTER TABLE
SELECT add_compression_policy('conditions', '1 week'::interval,
       schedule_interval => INTERVAL '1 minute') AS job_id \gset
SELECT pg_sleep(5);             --Just to let it start
 pg_sleep 
----------
 
(1 row)

SELECT job_id, application_name, next_start FROM timescaledb_information.jobs where job_id = :job_id;
 job_id |     application_name      |          next_start           
--------+---------------------------+-------------------------------
   1000 | Compression Policy [1000] | 2023-08-18 11:14:33.083837+02
(1 row)

SELECT pg_sleep(120);
 pg_sleep 
----------
 
(1 row)

SELECT job_id, application_name, next_start FROM timescaledb_information.jobs where job_id = :job_id;
 job_id |     application_name      |          next_start           
--------+---------------------------+-------------------------------
   1000 | Compression Policy [1000] | 2023-08-18 11:16:33.125928+02
(1 row)

SELECT alter_job(:job_id, scheduled => FALSE);
                                                                                     alter_job                                                                                      
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 (1000,00:01:00,00:00:00,-1,01:00:00,f,"{""hypertable_id"": 1, ""compress_after"": ""7 days""}","2023-08-18 11:16:33.125928+02",_timescaledb_internal.policy_compression_check,f,,)
(1 row)

SELECT alter_job(:job_id, scheduled=>TRUE, next_start=>now() + '10s'::interval);
                                                                                     alter_job                                                                                      
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 (1000,00:01:00,00:00:00,-1,01:00:00,t,"{""hypertable_id"": 1, ""compress_after"": ""7 days""}","2023-08-18 11:15:48.056363+02",_timescaledb_internal.policy_compression_check,f,,)
(1 row)

SELECT job_id, application_name, next_start FROM timescaledb_information.jobs where job_id = :job_id;
 job_id |     application_name      |          next_start           
--------+---------------------------+-------------------------------
   1000 | Compression Policy [1000] | 2023-08-18 11:15:48.056363+02
(1 row)

SELECT pg_sleep(20);
 pg_sleep 
----------
 
(1 row)

SELECT job_id, application_name, next_start FROM timescaledb_information.jobs where job_id = :job_id;
 job_id |     application_name      |          next_start           
--------+---------------------------+-------------------------------
   1000 | Compression Policy [1000] | 2023-08-18 11:16:48.085693+02
(1 row)
```

**Comment by RobAtticus** [comment]
Here are my latest steps to reproduce. Worked first time for me, but it may take a few tries of stopping the job mid-run to make it stuck:

Job def:
```
CREATE OR REPLACE PROCEDURE sleepy(IN job_id INT, IN config jsonb)
    LANGUAGE plpgsql
AS $procedure$
BEGIN
    PERFORM pg_sleep(120);
END;
$procedure$;
```

Add job and see its running:
```sql
tsdb=> SELECT add_job('sleepy', '1m');
tsdb=> SELECT * FROM pg_stat_activity WHERE application_name = 'User-Defined Action [1002]';
 datid | datname |  pid   | leader_pid | usesysid |  usename  |      application_name      | client_addr | client_hostname | client_port |         backend_start         |          xact_start           |          query_start          |        state_change         | wait_event_type | wait_event | state  | backend_xid | backend_xmin |       query_id       |        query         |        backend_type
-------+---------+--------+------------+----------+-----------+----------------------------+-------------+-----------------+-------------+-------------------------------+-------------------------------+-------------------------------+-----------------------------+-----------------+------------+--------+-------------+--------------+----------------------+----------------------+----------------------------
 16468 | tsdb    | 661492 |            |    16422 | tsdbadmin | User-Defined Action [1002] |             |                 |             | 2024-05-28 20:18:30.312978+00 | 2024-05-28 20:18:30.314624+00 | 2024-05-28 20:18:30.314624+00 | 2024-05-28 20:18:30.3173+00 | Timeout         | PgSleep    | active |             |         3479 | -3416356442043621232 | CALL public.sleepy() | User-Defined Action [1002]
(1 row)
```
stop job while running:
```sql
tsdb=> SELECT alter_job(1002, scheduled=>FALSE);
                                       alter_job
---------------------------------------------------------------------------------------
 (1002,00:01:00,00:00:00,-1,00:05:00,f,,-infinity,,t,"2024-05-28 20:18:29.168236+00",)
(1 row)

tsdb=> SELECT * FROM pg_stat_activity WHERE application_name = 'User-Defined Action [1002]';
 datid | datname | pid | leader_pid | usesysid | usename | application_name | client_addr | client_hostname | client_port | backend_start | xact_start | query_start | state_change | wait_event_type | wait_event | state | backend_xid | backend_xmin | query_id | query | backend_type
-------+---------+-----+------------+----------+---------+------------------+-------------+-----------------+-------------+---------------+------------+-------------+--------------+-----------------+------------+-------+-------------+--------------+----------+-------+--------------
(0 rows)
```
try to start again:
```sql
tsdb=> SELECT alter_job(1002, scheduled=>TRUE);
                                       alter_job
---------------------------------------------------------------------------------------
 (1002,00:01:00,00:00:00,-1,00:05:00,t,,-infinity,,t,"2024-05-28 20:18:29.168236+00",)
(1 row)
wait a bit, still not running:
tsdb=> SELECT * FROM pg_stat_activity WHERE application_name = 'User-Defined Action [1002]';
 datid | datname | pid | leader_pid | usesysid | usename | application_name | client_addr | client_hostname | client_port | backend_start | xact_start | query_start | state_change | wait_event_type | wait_event | state | backend_xid | backend_xmin | query_id | query | backend_type
-------+---------+-----+------------+----------+---------+------------------+-------------+-----------------+-------------+---------------+------------+-------------+--------------+-----------------+------------+-------+-------------+--------------+----------+-------+--------------
(0 rows)
```

**Comment by niconosenzo** [comment]
We might have found another service hitting this same issue (https://github.com/timescale/Support-Dev-Collab/issues/1792).
tl;dr, they hit an issue on May, 22nd affecting all the read queries, after that compression policies just stopped and afterwards the jobs didn't restart. I tried enabling/disabling them but are being silently ignored. 
tsdb 2.14.2
